src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5429,11 +5437,22 @@ public override BoundNode VisitDiscardExpression(BoundDiscardExpression node)
 
         public override BoundNode VisitThrowExpression(BoundThrowExpression node)
         {
-            var result = base.VisitThrowExpression(node);
+            var result = VisitRvalue(node.Expression);
+            CheckPossibleNullReceiver(node.Expression);
```

Comment:
>CheckPossibleNullReceiver [](start = 12, length = 25)

The use of this method doesn't look appropriate here. This expression is not a receiver.